### ISSUE TITLE
Support of 'h' -- sending/receiving of file descriptors

### ISSUE THREAD

**Issue by socketpair** [cocagne/txdbus#24]
When implementing, we should return file objects  `os.fdopen(received_fd)` in order not to lost FD on, say, exception. If someone wants to use file with another parameters, please use `.fileno()`.

One thing. CLOEXEC. https://sourceware.org/bugzilla/show_bug.cgi?id=12539 . This should be set by default as I think.

Also, suppose code:

```
with open(...) as f:
    # suppose f.fileno() = 42
    yield xxx.callRemote(f)
# (B)
f2 = open(...)
# f2.fileno() == 42 too
```

Due to #20 in some situations, we may go to point `(B)` while message is not sent actually (just buffered in twisted). But file `f` already closed. And more over, another file may be opened under the same descriptor. So, we should guarantee to wait for `sendmsg()` syscall in ANY cases.

**Comment by exvito** [comment]
@socketpair,

PR #42 implements UNIX_FD sending/receiving as method arguments and is getting ready for merging. You may want to take it for a spin, if this is still relevant to you.

Your feedback will be welcome.

Regarding your note on potentially sending the wrong FD: Your sample code uses `yield callRemote` which, assuming under `defer.inlineCallbacks`, will suspend execution until the underlying deferred fires (in other words: until the call goes through, is processed, and the response is successfuly received); only then will the `with open(...)` context be exited and the associated file (and FD) closed, after which `f2 = open(...)` runs. I can't see a problem with this code, under the assumption that it is indeed running in a `defer.inlineCallbacks` decorated function/method. Maybe I'm missing something you could point out?

**Comment by socketpair** [comment]
Sorry, I have lost interest for this library...

**Comment by exvito** [comment]
FYI, this was merged in today. Hopefully there'll soon be an updated release on PyPI.

**Comment by cocagne** [comment]
@exvito & @hobarrera let me know when you guys think you have everything in
shape for the next release and I'll cut the next version. Feel free to
update the metadata in setup.py accordingly.

**Comment by exvito** [comment]
Like I commented a while back, this feature is merged, tested and includes examples and is currently available in the master branch.

Should be available on PyPI once #64 completes.

Closing.
